From: Kemeng Shi <shikemeng@huaweicloud.com>
Subject: Xarray: remove repeat check in xas_squash_marks()
Date: Fri, 13 Dec 2024 20:25:22 +0800

Caller of xas_squash_marks() has ensured xas->xa_sibs is non-zero.  Just
remove repeat check of xas->xa_sibs in xas_squash_marks().

Link: https://lkml.kernel.org/r/20241213122523.12764-5-shikemeng@huaweicloud.com
Signed-off-by: Kemeng Shi <shikemeng@huaweicloud.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/xarray.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/lib/xarray.c~xarray-remove-repeat-check-in-xas_squash_marks
+++ a/lib/xarray.c
@@ -128,9 +128,6 @@ static void xas_squash_marks(const struc
 	unsigned int mark = 0;
 	unsigned int limit = xas->xa_offset + xas->xa_sibs + 1;
 
-	if (!xas->xa_sibs)
-		return;
-
 	do {
 		unsigned long *marks = xas->xa_node->marks[mark];
 		if (find_next_bit(marks, limit, xas->xa_offset + 1) == limit)
_
